### PR TITLE
fix(2313): Get Github binding before trying to set it [2]

### PR DESCRIPTION
**Pull request by tkyi**
## Context

The automatic Github binding seems to be flaky. It's hard to parse through logs when multiple calls are made to the same projects to set the binding. 

## Objective

The above issue would be easier to debug with fewer logs and calls to the Sonar API.
This PR checks if the Github binding exists and only tries to set it if it doesn't. Also takes in additional params to getAccessToken to construct the proper projectName.

## References

Related to https://github.com/screwdriver-cd/screwdriver/issues/2313

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -79,26 +79,41 @@ function configureGitApp(projectKey, projectName) {
...
-        uri: `${sonarHost}/api/alm_settings/set_github_binding?${parameters}`,
+        method: 'GET',
+        uri: `${sonarHost}/api/alm_settings/get_binding?project=${componentId}`,
         auth: {
             username: adminToken
         }
-    }).catch((err) => {
-        // if cannot configure app, do not throw err
+    }).catch(() => {
+        console.log('in catch!!!');
```
Comment (pritamstyz4ever) [inline index.js:92]: remove debug log may be?

File: index.js [hunk 2]
```diff
@@ -332,21 +347,27 @@ class CoverageSonar extends CoverageBase {
...
      *                                          to talk to coverage server
      */
-    _getAccessToken({ scope, username, projectKey, projectName, buildCredentials }) {
+    _getAccessToken({ scope, username, projectKey, projectName, jobName, pipelineName,
+        buildCredentials }) {
         const { jobId, pipelineId, prParentJobId } = buildCredentials;
         let projectData = { username, projectKey, projectName };
 
-        if (!username || !projectKey || !projectName) {
+        if (!username || !projectKey || !projectName || projectName.includes('undefined')) {
```
Comment (pritamstyz4ever) [inline index.js:363]: if projectName is a string should we be checking 
`typeof(projectName) == "undefined" `

Comment (tkyi) [reply index.js:363]: ah no was checking literal inclusion of string "undefined"

**Review by jithine — APPROVED** [review]
